bigframes/session/dry_runs.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import copy
+from typing import Any, Dict, List, Sequence
+
+from google.cloud import bigquery
+import pandas
+
+from bigframes import dtypes
+
+
+def get_table_stats(table: bigquery.Table) -> pandas.Series:
+    values: List[Any] = []
+    index: List[Any] = []
+
+    # Indicate that no query is executed.
+    index.append("isQuery")
+    values.append(False)
+
+    # Populate column and index types
+    col_dtypes = dtypes.bf_type_from_type_kind(table.schema)
+    index.append("tableColumnCount")
+    values.append(len(col_dtypes))
+    index.append("tableColumnTypes")
```

Comment:
It's not super easy for end user too predict if something will result in a query or just read the table directly. Could we try to align these names so that they don't need as much logic to handle one case over the other?

Reply:
SG. Done